src/app/(drawerPages)/badges/[id]/page.tsx

```diff
@@ -45,61 +60,143 @@ export default function BadgeDetails({ params }: Props) {
     }
   }, [isDisconnected, router]);
 
+  useEffect(() => {
+    drawerStates.badgeActions.bottom = true;
+  }, [currentBadge, drawerStates.badgeActions]);
+
+  const ctaButton = useMemo(() => {
+    if (isOwned) return null;
+    if (!currentBadge) return null;
+    if (currentBadge.type === BadgeTypeEnum.IRL) {
+      return null;
```

Comment:
/nit you can combine these condition since they all return null
```suggestion
    if (isOwned || !currentBadge || currentBadge.type === BadgeTypeEnum.IRL) 
    {
        return null;
    }
```

Reply:
done